q2_moshpit/busco/tests/test_busco.py

```diff
@@ -0,0 +1,359 @@
+        that are supposed to be generated. These files are empty.
+
+        Args:
+            tmp_path (str): path where to write empty files
+            bins (MultiMAGSequencesDirFmt): fixture from which the sample ids
+                are extracted.
+        """
+        # Get output path
+        output_dir = os.path.join(tmp_path, "busco_output")
+
+        # Creates pandas df "manifest" from bins
+        manifest: pd.DataFrame = bins.manifest.view(pd.DataFrame)
+
+        # numpy.ndarray with unique dirs
+        sample_ids = manifest.index.get_level_values(0).unique()
+
+        # write files to output
+        # Loop to create the empty files
+        expected = {}
+        for sample_id in sample_ids:
+            # Specify the name of each empty file
+            path_to_run_summary = os.path.join(
+                output_dir, sample_id, "batch_summary.txt"
+            )
+            os.makedirs(
+                os.path.dirname(path_to_run_summary),
+                exist_ok=True
+            )
+
+            # Save to dict
+            expected[sample_id] = path_to_run_summary
+
+            # Create an empty file
+            with open(path_to_run_summary, 'w'):
+                pass
+
+        # Return expected output
+        return expected
+
+    @patch('subprocess.run')
+    def test_run_busco(self, subp_run):
+        """
+        Test function `_run_busco`. Checks for dictionary equality.
+        """
+        with tempfile.TemporaryDirectory() as tmp_path:
+            # Define command arguments and parse them
+            fake_props = {"a": "b", "c": "d"}
+            expected = self.mock_run_busco(
+                tmp_path=tmp_path, bins=self.get_dummy_mags()
+            )
+
+            # Run busco and save paths to run summaries
+            observed = _run_busco(
+                output_dir=os.path.join(tmp_path, "busco_output"),
+                mags=self.get_dummy_mags(),
+                params=fake_props,
+            )
+
+            self.assertDictEqual(expected, observed)
+
+    @patch("subprocess.run")
+    def test_run_busco_exception(self, subp_run):
+        """
+        Test function `_run_busco`. Checks for a raised exception.
+        """
+        with tempfile.TemporaryDirectory() as tmp_path:
+            # Define command arguments and parse them
+            fake_props = {"a": "b", "c": "d"}
+
+            with self.assertRaises(FileNotFoundError):
+                # Run busco and save paths to run summaries
+                _ = _run_busco(
+                    output_dir=os.path.join(tmp_path, "busco_output"),
+                    mags=self.get_dummy_mags(),
+                    params=fake_props,
+                )
+
+    # Integration test busco.
+    @patch('q2_moshpit.busco.utils._run_busco')
+    @patch('q2_moshpit.busco.utils._zip_busco_plots')
+    @patch('q2_moshpit.busco.utils._draw_busco_plots')
+    @patch('q2_moshpit.busco.utils._collect_summaries_and_save')
```

Comment:
Same as above: you should add assertions that all of those mocks were called with the arguments you expect them to (or at least those where it's possible).